### PR TITLE
Adding Windows Server 2022 and cleaning up windows drone images.

### PR DESCRIPTION
**Pull request by phillipsj**
Adds 2022 support and cleans up drone windows images. 

Signed-off-by: Jamie Phillips <jamie.phillips@suse.com>

### CODE REVIEW

**Review by luthermonson — APPROVED** [review]

File: manifest-go115.tmpl [hunk 1]
```diff
@@ -11,4 +11,10 @@ manifests:
...
       os: windows
-      version: 20H2
\ No newline at end of file
+      version: 20H2
+  -
+    image: rancher/golang:1.15-windowsservercore-ltsc2022
+    platform:
+      architecture: amd64
+      os: windows
+      version: ltsc2022
```
Comment (rosskirkpat) [inline manifest-go115.tmpl]: Did you mean to add 2022 support to go1.15? I think we should stick with 1.16 only until we add 1.17 and then have 1.16 and 1.17 be the only versions that have 2022 builds

**Review by rosskirkpat — CHANGES_REQUESTED** [review]

File: manifest-go115.tmpl (hunk 1 above)
Comment (luthermonson) [reply manifest-go115.tmpl]: not sure this really hurts anything building this tho

**Comment by phillipsj** [comment]
Linking up why this is being added

* https://github.com/rancher/windows/issues/104

**Review by rosskirkpat — APPROVED** [review]
LGTM!

**Pull request merged**